design/one-pager-easy-stacks.md

```diff
@@ -0,0 +1,294 @@
+an external system, such as a cloud provider. _External resources_ in the
+provider's API are modelled as _managed resources_ in the Kubernetes API server.
+Managed resources are considered the domain of _infrastructure operators_;
+they're cluster scoped infrastructure like a `Node` or `PersistentVolume`.
+_Application operators_ may claim a managed resource for a particular purpose by
+creating a namespaced _resource claim_. Managed resources may be provisioned
+explicitly before claim time (static provisioning), or automatically at claim
+time (dynamic provisioning). The initial configuration of dynamically
+provisioned managed resources is specified by a _resource class_.
+
+In most of the cases, the resources have to refer to each other in order to be
+connected at the cloud provider level. For example, let's say you have an application
+like Wordpress that you want to deploy to a Kubernetes cluster and have its database
+to be managed by the cloud provider. While you can connect from your cluster to
+the database over the internet, it's usually desirable to have both of them in a
+private network to connect to each other and expose only your application's IP
+to the world. There are usually a few resources that you need to create to achieve
+this like VPC, sub-network, internet gateway etc. After creating these resources,
+you need to refer to them in your resource classes that you author for database
+and cluster claims to use.
+
+Today, it's possible to do all this by manually configuring each resource YAML
+files and run `kubectl apply -f <directory of YAMLs>`. However, there are two areas
+that we can improve during this process:
+* The preparation phase of these YAMLs requires to be very careful and changing
+  something usually means touching a few different files. While this may not seem
+  like a big hurdle, changing some high level parameters of that environment is
+  cumbersome _after_ it's deployed.
+* A user who is familiar with AWS may not be familiar with Azure or GCP; so,
+  ready-made configuration stacks that promise the exact same setup would be
+  useful for them to see how they can achieve similar setups and run various
+  benchmarks according to their business needs.
+* After you run `kubectl apply -f <directory of YAMLs>`, a dependency graph of
+  resources starts to resolve. However, it's not always clear for user to know _when_
+  everything is actually ready to use unless they know readiness of which resources
+  signal the readiness of the whole environment.
+
+## Goals
+
+The main goal of this design is to make it easier for people to write sets of
+configurations and be able to manage those configuration sets on a higher level.
+
+It is important that the design puts forward:
+
+* A base boilerplate package where creating a new configuration set that has a
+  controller is as easy as changing the YAML files.
+* A way to declare pre-requisites that needs to be satisfied by the cluster
```

Comment:
What is the benefit of including this in the design? I would not expect a formal mechanism to be needed at this point, since people can just install things in the right order according to some documentation. Is this referring to listing the crds it needs to be able to CRUD?

Reply:
I was under the misconception that the stack `app.yaml` actually provided that ability. But I see that's not the case https://github.com/crossplaneio/crossplane/pull/1110#discussion_r352728755

So, I'll just remove that and expect people to have installed all the CRDs needed.